ui/shared/components/panel/acmg/AcmgCriteria.jsx

```diff
@@ -29,16 +44,31 @@ export const getNewScoreValue = (criteria) => {
 class AcmgCriteria extends React.PureComponent {
 
   static propTypes = {
-    criteria: PropTypes.arrayOf(PropTypes.string).isRequired,
-    setCriteria: PropTypes.func.isRequired,
-    acmgClassification: PropTypes.string.isRequired,
-    setAcmgClassification: PropTypes.func.isRequired,
-    setActive: PropTypes.func.isRequired,
+    setInactive: PropTypes.func.isRequired,
     variant: PropTypes.object.isRequired,
     dispatchUpdateVariantClassification: PropTypes.func,
   }
 
-  state = { formWarning: '' }
+  constructor(props) {
+    super(props)
+
+    const { variant } = this.props
+    const { classify = 'Unknown', criteria = [] } = variant.acmgClassification || {}
+    // eslint-disable-next-line react/state-in-constructor
+    this.state = { formWarning: '', criteria, classify, score: getNewScoreValue(criteria) }
```

Comment:
I'm curious about how to meet the eslint requirement of no 'react/state-in-constructor' in this case. Can we define the states outside of the constructor? And assign initial values to them in the `componentDidMount` function?

Reply:
no thats terrible practice in React and super prone to bugs. The linter rule encourages us not to use props to initialize state because thats better practice, but for use cases like this where its substantially better code if we do, its fine to disable the linter